syntax = "proto3";

option go_package = "services/proto";

import "services/proto/s2s_follow.proto";

message Approval {
  FollowDetails follow = 1;
  bool accept = 2;
}

message ReceivedApproval {
  ReceivedFollowDetails follow = 1;
  string accept = 3;
}

message ApprovalResponse {
  enum ResultType {
    OK = 0;
    ERROR = 1;
  }

  ResultType result_type = 1;

  /* Should only be set if result_type is not OK. */
  string error = 2;
}

// Service for sending and receiving server-to-server follow requests.
service Approver {
  rpc SendApproval(Approval) returns (ApprovalResponse);
  rpc ReceiveApproval(ReceivedApproval) returns (ApprovalResponse);
}
